src/source/host_git_source.rs

```diff
@@ -179,7 +179,7 @@ pub fn git_src(
         return Err(SourceError::GitErrorStr("failed to git reset"));
     }
 
-    if git_lfs_required(&cache_path) {
+    if source.lfs() || git_lfs_required(&cache_path) {
```

Comment:
Shouldn't this be a `&&`? I think the explicit setting should be required to enable LFS and we could even skip the lfs required check (or run it just to save some extra invocations if it's not actually required).

Reply:
makes sense